### ISSUE TITLE
Hang when using Find with PDF open

### ISSUE THREAD

**Issue by ASM-SW** [MicrosoftEdge/WebView2Feedback#1889]
I am testing an application that uses ICoreWebView2ExperimentalSettings6 to hide the PDF controls.  I have a Windows program that is based on https://docs.microsoft.com/en-us/microsoft-edge/webview2/get-started/win32.  I am testing with:
  Microsoft.Web.Webview2 version:  1.0.1018-prerelease
  I reproduced this with the following version of the edge browser:
  fixed binary edge 95.01020.30 
  Dev version 96.0.1054.2
  Canary edge:  97.0.1059.0

I navigate to a PDF file on my computer.  I click on the magnifying class to bring up the find window.  I move my mouse between the  x to close the find windows and the x to close the program.  Within less that half a minute the program hangs.  I can see this because when I move the mouse around the x to close either the find window or the program no longer gets highlighted.

The command I make to hide toolbar items is:

```
	ICoreWebView2Settings* Settings;
	webviewWindow->get_Settings(&Settings);
	Settings->put_AreDefaultContextMenusEnabled(FALSE);
	Settings->put_AreDefaultScriptDialogsEnabled(TRUE);
	Settings->put_AreDevToolsEnabled(FALSE);
	Settings->put_AreHostObjectsAllowed(TRUE);
	//Disable the built in error page because it shows the URL : localhost
	//It also shows instructions to items that the user cannot access.These items such as proxy, firewall and antivirus.
	Settings->put_IsBuiltInErrorPageEnabled(FALSE);
	Settings->put_IsScriptEnabled(TRUE);
	Settings->put_IsStatusBarEnabled(FALSE);
	Settings->put_IsWebMessageEnabled(TRUE);
	Settings->put_IsZoomControlEnabled(FALSE);

	// Resize WebView to fit the bounds of the parent window
	RECT bounds;
	GetClientRect(hWnd, &bounds);
	sWebviewController->put_Bounds(bounds);

	// Setup PDF viewer toolbar, hide buttons that create security issues like print and save
	wil::com_ptr<ICoreWebView2ExperimentalSettings6> experimentalSettings6;
	HRESULT hr = Settings->QueryInterface(&experimentalSettings6);
	hr = experimentalSettings6->put_HiddenPdfToolbarItems(COREWEBVIEW2_PDF_TOOLBAR_ITEMS_SAVE | COREWEBVIEW2_PDF_TOOLBAR_ITEMS_PRINT | COREWEBVIEW2_PDF_TOOLBAR_ITEMS_SAVE_AS);
```

[AB#36927919](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/36927919)

**Comment by champnic** [comment]
Thanks for the bug report @ASM-SW - I've added it to our backlog to take a look!

**Comment by ASM-SW** [comment]
I ran a few more tests today:  
- repeated with latest canary build:  97.0.1065.0, still got the hang
- I tried not hiding the PDF toolbar items:  still got the hang
- I opened my PDF build in the edge canary browser and did not get the hang.

**Comment by champnic** [comment]
Thanks for the extra details. 

>I tried not hiding the PDF toolbar items: still got the hang

Sounds like it's related to Find + PDF, rather than the PDF toolbar items setting. I'll update the title.

**Comment by ASM-SW** [comment]
Hi,

I ran this from within WinDbg to get more information for you.  After it hang I ran the following commands that we use for collecting hang information.
```
.sympath C:\dev\CustomPDFViewer\x64\Debug;symsrv*symsrv.dll*c:\dev\symbols*http://msdl.microsoft.com/download/symbols
~0 s
~*kP
!analyze -hang
```
The attached file is the output of the command window.  I hope this will help you.
[WindbgOutput.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/7477140/WindbgOutput.txt)

After running the above commands I entered g to return to the program.  It was still hung.  After uploading the file.  I checked and the program was no longer hanging.

**Comment by champnic** [comment]
@ASM-SW I saw you closed the issue - is this no longer a problem for you?

**Comment by ASM-SW** [comment]
I did not mean to close.  I must have clicked on the wrong button.

**Comment by ASM-SW** [comment]
Have you been able to duplicate this issue?

**Comment by champnic** [comment]
We have not had the chance to investigate this issue yet.

**Comment by mayuragarwal-qm** [comment]
I am also facing similar issue when tried with runtime version 105.0.1343.53
Any updates on the fix?

**Comment by johna-ms** [comment]
Hi we were able to duplicate this issue by opening find bar and clicking near it but over app content ie. non webview content. I have a workaround for this pending that will causes those clicks to go to the find bar instead of hanging the app. Hopefully this resolves the issue. Work for a more complete fix is planned at a lower priority.

**Comment by mayuragarwal-qm** [comment]
This issue happens not just during find in PDF but even using ctrl + F in a normal webview2 browser screen.
WPF application hangs searching on a page with WebView2.

**Comment by ASM-SW** [comment]
We switched to Sumatra PDF.  It has a built in kiosk mode that worked for us with a smaller footprint.

**Comment by champnic** [comment]
Going to keep this reopen until we complete the fix.

**Comment by champnic** [comment]
This should be fixed in runtimes 107.0.1418.11+. Thanks!

**Comment by ASM-SW** [comment]
Thank you
